pkg/scalers/gcp_pubsub_scaler.go

```diff
@@ -145,14 +145,14 @@ func (s *pubsubScaler) IsActive(ctx context.Context) (bool, error) {
 			s.logger.Error(err, "error getting Active Status")
 			return false, err
 		}
-		return size > s.metadata.activationValue, nil
+		return size > float64(s.metadata.activationValue), nil
 	case pubsubModeOldestUnackedMessageAge:
 		delay, err := s.getMetrics(ctx, pubSubStackDriverOldestUnackedMessageAgeMetricName)
 		if err != nil {
 			s.logger.Error(err, "error getting Active Status")
 			return false, err
 		}
-		return delay > s.metadata.activationValue, nil
+		return delay > float64(s.metadata.activationValue), nil
```

Comment:
same as above

Reply:
The metadata parsing has been updated and removed the cast to float64.